rtabi32/rtabi32.rst

```diff
@@ -0,0 +1,1863 @@
+   contains two 4-byte fields, the element size followed by the element
+   count.
+
+Below we list the functions and their arguments. For details see the
+references cited at the start of `Construction and destruction of arrays`_.
+
+.. code-block:: c
+
+  void *__cxa_vec_new(
+      size_t count, size_t element_size, size_t cookie_size,
+          void (*ctor)(void *), void (dtor)(void *));
+  void *__cxa_vec_new2(
+       size_t count, size_t element_size, size_t cookie_size,
+           void (*ctor)(void *this), void (*dtor)(void *this),
+               void *(*alloc)(size_t size), void (*dealloc)(void *object));
+  void *__cxa_vec_new3(		
+      size_t count, size_t element_size, size_t cookie_size,
+          void (*ctor)(void *this), void (*dtor)(void *this),
+              void *(*alloc)(size_t size), void (*dealloc)(void *object, size_t size));
+  void *__cxa_vec_ctor(
+      void *vector, size_t count, size_t element_size,
+          void (*ctor)(void *this),  void (*dtor)(void *this));
+  void __cxa_vec_dtor(
+      void *vector, size_t count, size_t element_size,
+          void (*dtor)(void *this));
+  void __cxa_vec_cleanup(
+      void *vector, size_t count, size_t element_size,
+          void (*dtor)(void *this));
+  void __cxa_vec_delete(
+      void *vector, size_t element_size, size_t cookie_size,
+          void (*dtor)(void *this));
+  void __cxa_vec_delete2(
+      void *vector, size_t element_size, size_t cookie_size,
+          void (*dtor)(void *this),
+              void (*dealloc)(void *object));
+  void __cxa_vec_delete3(
+      void *vector, size_t element_size, size_t cookie_size,
+          void (*dtor)(void *this),
+              void (*dealloc)(void *object, size_t size));
+  void *__cxa_vec_cctor(
+      void *destination, void *source, size_t count, size_t element_size, 
+          void (*copy_ctor)(void *this, void *source),
+              void (*dtor)(void *this));
+
+Helper functions defined by the C++ ABI for the Arm Architecture
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This ABI define the following new helpers which can be called more
+efficiently.
+
+.. code-block::
+
+      __aeabi_vec_ctor_nocookie_nodtor
+      __aeabi_vec_ctor_cookie_nodtor
+      __aeabi_vec_cctor_nocookie_nodtor
+      __aeabi_vec_new_cookie_noctor
+      __aeabi_vec_new_nocookie
+      __aeabi_vec_new_cookie_nodtor
+      __aeabi_vec_new_cookie
+      __aeabi_vec_dtor
+      __aeabi_vec_dtor_cookie
+      __aeabi_vec_delete
+      __aeabi_vec_delete3
+      __aeabi_vec_delete3_nodtor
+      __aeabi_atexit
+
+Compilers are not required to use these functions but runtime
+libraries complying with this ABI must supply them. Below we list the
+functions and their arguments. For details see CPPABI32_ section
+'Array construction and destruction'. Each function is declared extern
+“C”.
+
+.. code-block:: c
+
+  void* __aeabi_vec_ctor_nocookie_nodtor(
```

Comment:
Is it me, or is this some obvious indentation convention I'm not aware of?

Reply:
The thing where you put the open paren at the end of the line instead of first thing on the next? I've always thought of that as Python-inspired (an unclosed pair of parens stops the parser from treating the newline as statement-terminating), though I see it reasonably often in C/C++ as well these days.

Reply:
No, I think that's an accepted C idiom. I mean the tiered indentation of the arguments below this line. Each new line is indented 4 more spaces to the right.

Reply:
Ah, I was misled by Github not actually showing those lines above your comment!

No, I don't think that's any convention I recognise. My guess would be that it got mangled at some point in the past by pasting already-indented text into an auto-indenting editor. I've realigned it all in the revised version I just force-pushed.